cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -121,7 +123,8 @@ public AppFabricServer(CConfiguration cConf, SConfiguration sConf,
                          PluginService pluginService,
                          PrivilegesFetcherProxyService privilegesFetcherProxyService,
                          @Nullable AppVersionUpgradeService appVersionUpgradeService,
-                         RouteStore routeStore) {
+                         RouteStore routeStore,
+                         CoreSchedulerService coreSchedulerService) {
```

Comment:
This won't be a singleton injection. This is because in the guice module, the singleton binding was on the `Scheduler`. When requesting a `CoreSchedulerService` in here, it is using JIT binding instead. 

You can either inject `Scheduler` here and do `scheduler instanceof Service` to optionally start/stop it (which has slightly better abstraction) or add a singleton binding for the `CoreSchedulerService` in the guice module by 

```java
bind(CoreSchedulerService.class).in(Scopes.SINGLETON);
bind(Scheduler.class).to(CoreSchedulerService.class);
```